### DISCUSSION TITLE
Is Metals able to keep two or more projects loaded at the same time?

### DISCUSSION THREAD

**Discussion by caenrique** [scalameta/nvim-metals#469] [General]
Context: I want to switch between different scala projects without leaving nvim or opening multiple nvim instances

I'll try to describe the problem using project A and B:

- open project A -> metals attach and imports the project
- open project B by changing the cwd -> metals seems to import the project fine and keeps working
- open project A again by changing the cwd -> metals no longer works and shuts down the build server

To work around this issue, I'm just restarting metals every time I change projects, but that also means reimporting the whole project.

I was hoping that metals could keep both projects imported at the same time and keep track of the cwd or something.

Is this something already solved? Has someone tried to do something similar with a better approach than me? 😄

**Comment by ckipp01** [comment]
Good question @caenrique. Honestly, I don't have a great answer. The short answer is that no, Metals can only correctly handle one valid workspace at a time. My only recommendation would be to have two different instances running,  one for each project, but that's probably not what you want.
